x-pack/filebeat/input/awss3/input.go

```diff
@@ -376,5 +380,25 @@ func getProviderFromDomain(endpoint string, ProviderOverride string) string {
 	return "unknown"
 }
 
+func pollSqsWaitingMetric(ctx context.Context, receiver *sqsReader) {
+	t := time.NewTicker(time.Minute)
+	defer t.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			count := receiver.GetApproximateMessageCount(ctx)
+			if count == -1 {
+				// stop polling if error is encountered
```

Comment:
Could you make this stop only if the client does not have permissions to make the call. If there's a connectivity issue while this is running (for example) we don't want this thing to stop reporting the metric.